CMakeLists.txt

```diff
@@ -47,34 +47,115 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
-    ${SIMDxorshift_SOURCE_DIR}/src/xorshift128plus.c)
-  list(APPEND VAMP_INCLUDES ${SIMDxorshift_SOURCE_DIR}/include)
 endif()
 
-list(APPEND VAMP_INCLUDES
+# Create VAMP C++ library
+add_library(vamp_cpp INTERFACE)
+
+# Always use only the subdirectory for install interface to avoid CMake's parent directory bug
+# See: https://gitlab.kitware.com/cmake/cmake/-/issues/17950 and related
+# Do NOT use INCLUDES DESTINATION in install(TARGETS ...) for INTERFACE libraries
+# If you change include structure, update this path accordingly
+
+# Split common includes into exportable and non-exportable parts
+set(VAMP_EXPORTABLE_INCLUDES
   ${NIGH_INCLUDE_DIRS}
   ${PDQSORT_INCLUDE_DIRS}
-  src/impl
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/impl
+)
+
+set(VAMP_BUILD_ONLY_INCLUDES)
+if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64" AND SIMDxorshift_SOURCE_DIR)
+  list(APPEND VAMP_BUILD_ONLY_INCLUDES ${SIMDxorshift_SOURCE_DIR}/include)
+endif()
+
+# Common include directories for both C++ library and Python bindings
+set(VAMP_COMMON_INCLUDES ${VAMP_EXPORTABLE_INCLUDES} ${VAMP_BUILD_ONLY_INCLUDES})
+
+message(STATUS "VAMP_COMMON_INCLUDES: ${VAMP_COMMON_INCLUDES}")
+
+target_include_directories(vamp_cpp
+    INTERFACE
+        $<BUILD_INTERFACE:${VAMP_EXPORTABLE_INCLUDES}>
+        $<BUILD_INTERFACE:${VAMP_BUILD_ONLY_INCLUDES}>
+        $<INSTALL_INTERFACE:include/vamp>
 )
 
-if(VAMP_BUILD_PYTHON)
-  find_package(Python 3.8
-    REQUIRED COMPONENTS Interpreter Development.Module
-    OPTIONAL_COMPONENTS Development.SABIModule)
+target_link_libraries(vamp_cpp
+    INTERFACE
+    Eigen3::Eigen
+    nigh
+    pdqsort
+)
+
+# Create SIMDxorshift static library for x86_64 systems
+if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64" AND SIMDxorshift_SOURCE_DIR)
+  add_library(vamp_simd STATIC
+    ${SIMDxorshift_SOURCE_DIR}/src/simdxorshift128plus.c
+    ${SIMDxorshift_SOURCE_DIR}/src/xorshift128plus.c
+  )
+  
+  target_include_directories(vamp_simd
+    PRIVATE
+    ${SIMDxorshift_SOURCE_DIR}/include
+  )
+  
+  # Link the SIMD library to vamp_cpp
+  target_link_libraries(vamp_cpp
+    INTERFACE
+    vamp_simd
+  )
+  
+  # Set library properties for vamp_simd
+  set_target_properties(vamp_simd PROPERTIES
+    VERSION ${PROJECT_VERSION}
+    SOVERSION ${PROJECT_VERSION_MAJOR}
+  )
+endif()
 
-  CPMAddPackage("gh:wjakob/nanobind#358d452c314dbe8c07026d984ad8d5aa860f26fb")
+# Set library properties
+set_target_properties(vamp_cpp PROPERTIES
+    VERSION ${PROJECT_VERSION}
+    SOVERSION ${PROJECT_VERSION_MAJOR}
+)
+
+# Install C++ library headers and SIMD library (if available)
+set(VAMP_INSTALL_TARGETS vamp_cpp)
+if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64" AND TARGET vamp_simd)
+  list(APPEND VAMP_INSTALL_TARGETS vamp_simd)
+endif()
+
+install(TARGETS ${VAMP_INSTALL_TARGETS}
+    EXPORT vamp_cppTargets
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib
+    RUNTIME DESTINATION bin
+)
+
+# Install headers
+install(DIRECTORY src/impl/vamp/
```

Comment:
Just a note for the future: we should really pull `src/impl/vamp` out into `include/vamp` in the main project, for clearer layout.